From 6b77292e2064e0e1fa00ceaab73b9171a38a1613 Mon Sep 17 00:00:00 2001
From: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Date: Wed, 30 Jun 2021 06:23:17 +0300
Subject: [PATCH 24/29] Revert "s32gen1:pcie: Add dma-coherent to Linux dts"

Will move this statically in the Linux device-tree, as we intend for
U-Boot to no longer override the PCIe coherency settings.

This reverts commit 13db5bba2a7ea5e9b6bcfc843a1f15c927668006.

Upstream-Status: Pending

Signed-off-by: Bogdan Hamciuc <bogdan.hamciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
Issue: ALB-7389
---
 board/freescale/s32-gen1/common.c | 31 -------------------------------
 1 file changed, 31 deletions(-)

diff --git a/board/freescale/s32-gen1/common.c b/board/freescale/s32-gen1/common.c
index 292c1e96ea..45c673b51d 100644
--- a/board/freescale/s32-gen1/common.c
+++ b/board/freescale/s32-gen1/common.c
@@ -78,33 +78,6 @@ int checkboard(void)
 	return 0;
 }
 
-#if CONFIG_IS_ENABLED(PCIE_S32GEN1)
-static void ft_pcie_fixup(void *fdt)
-{
-	const char * const aliases[] = {"pcie0", "pcie1"};
-	const char *path;
-	int off = -1;
-	size_t i;
-
-	/* Add dma-coherent property to each PCIe node */
-	for (i = 0; i < ARRAY_SIZE(aliases); i++) {
-		path = fdt_get_alias(fdt, aliases[i]);
-		if (!path) {
-			pr_err("%s: No such alias: %s\n", __func__, aliases[i]);
-			break;
-		}
-
-		off = fdt_path_offset(fdt, path);
-		if (off < 0) {
-			pr_err("%s: Could not find path %s\n", __func__, path);
-			break;
-		}
-
-		if (fdt_setprop(fdt, off, "dma-coherent", NULL, 0))
-			pr_err("PCIe%ld fdt: could not set dma-coherent\n", i);
-	}
-}
-#endif
 
 #if defined(CONFIG_OF_FDT) && defined(CONFIG_OF_BOARD_SETUP)
 int ft_board_setup(void *blob, bd_t *bd)
@@ -115,10 +88,6 @@ int ft_board_setup(void *blob, bd_t *bd)
 	ft_enet_fixup(blob);
 #endif
 
-#if CONFIG_IS_ENABLED(PCIE_S32GEN1)
-	ft_pcie_fixup(blob);
-#endif
-
 	return 0;
 }
 #endif /* defined(CONFIG_OF_FDT) && defined(CONFIG_OF_BOARD_SETUP) */
-- 
2.17.1

